guides/source/upgrading_ruby_on_rails.md

```diff
@@ -90,6 +90,77 @@ size of the `bootsnap` cache for the others.
 
 Application controllers that inherit from `ActiveStorage::BaseController` and use streaming to implement custom file serving logic must now explicitly include the `ActiveStorage::Streaming` module. 
 
+### New `ActiveSupport::MessageEncryptor` default serializer
+
+As of Rails 7.1, the default serializer in use by the `MessageEncryptor` is `JSON`.
+This offers a more secure alternative to the current default serializer.
+
+The `MessageEncryptor` offers the ability to migrate the default serializer from `Marshal` to `JSON`.
+
+If you would like to ignore this change in existing applications, set the following: `config.active_support.default_message_encryptor_serializer = :marshal`.
+
+In order to roll out the new default when upgrading from `7.0` to `7.1`, there are three configuration variables to keep in mind.
+```
+config.active_support.default_message_encryptor_serializer
+config.active_support.fallback_to_marshal_deserialization
+config.active_support.use_marshal_serialization
+```
+
+`default_message_encryptor_serializer` defaults to `:json` as of `7.1` but it offers both a `:hybrid` and `:marshal` option.
+
+In order to migrate an older deployment to `:json`, first ensure that the `default_message_encryptor_serializer` is set to `:marshal`.
+```ruby
+# config/application.rb
+config.load_defaults 7.0
+config.active_support.default_message_encryptor_serializer = :marshal
+```
+
+Once this is deployed on all Rails processes, set `default_message_encryptor_serializer` to `:hybrid` to begin using the
+`ActiveSupport::JsonWithMarshalFallback` class as the serializer. The defaults for this class are to use `Marshal`
+as the serializer and to allow the deserialisation of both `Marshal` and `JSON` serialized payloads.
+
+```ruby
+config.load_defaults 7.0
+config.active_support.default_message_encryptor_serializer = :hybrid
+```
+
+Once this is deployed on all Rails processes, set the following configuration options in order to stop the
+`ActiveSupport::JsonWithMarshalFallback` class from using `Marshal` to serialize new payloads.
+
+```ruby
+# config/application.rb
+config.load_defaults 7.0
+config.active_support.default_message_encryptor_serializer = :hybrid
+config.active_support.use_marshal_serialization = false
+```
+
+Allow this configuration to run on all processes for a considerable amount of time.
+Once all `MessageEncryptor` payloads in transit are `JSON` serialized, the following configuration 
+options will disable the Marshal fallback in `ActiveSupport::JsonWithMarshalFallback`.
```

Comment:
Can we offer any advice about how to be confident that all messages are JSON-serialized?

I suppose if the application uses `expires_in` / `expires_at` for all messages, then there should be a clear point at which no valid Marshal-serialized messages remain. But perhaps it would also be helpful for `JsonWithMarshalFallback` to log fallbacks?

Reply:
Yeah that is a good point, we could likely solve this with some logging in `ActiveSupport::JsonWithMarshalFallback`

Reply:
Now that logging is in place, I think it's worth adding a sentence or two about it here. Maybe something like this:

> Allow this configuration to run on all processes for some time. Each time a `Marshal` serialized message is decrypted, a warning will be logged (`JsonWithMarshalFallback: Marshal load fallback occured.`). Once these warnings disappear and you are confident that all `MessageEncryptor` payloads in transit are `JSON` serialized, the following configuration options will disable the Marshal fallback in `ActiveSupport::JsonWithMarshalFallback`:

Reply:
Done, I included a bit about the fact that the logging is present to help determine when you're safe to move forward. Thanks!